integration-tests/src/test/java/org/openmrs/maven/plugins/SetupIntegrationTest.java

```diff
@@ -90,4 +90,15 @@ public void setup_shouldInstallServerFromDistroPropertiesDir() throws Exception{
         assertFilePresent(serverId+"/modules");
         assertModulesInstalled(serverId, "owa-1.4.omod", "uicommons-1.7.omod", "uiframework-3.6.omod");
     }
+
+    @Test
+    public void setup_shouldInstallServerWithGivenJdkAndCreateSdkPropertiesFileIfNotExist() throws Exception{
+        String serverId = UUID.randomUUID().toString();
+        addTaskParam("serverId", serverId);
+
+        executeTask("setup");
+        assertSuccess();
+        assertServerInstalled(serverId);
```

Comment:
You should actually check here that java.home was set or wasn't set in openmrs-server.properties...